examples/tv-casting-app/APIs.md

```diff
@@ -0,0 +1,680 @@
+Commissioner. In the context of the
+[Matter Video Player architecture](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/media/VideoPlayerArchitecture.adoc),
+a `CastingPlayer` would map to
+[Casting "Video" Player](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/media/VideoPlayerArchitecture.adoc#1-introduction).
+The `CastingPlayer` is expected to be hosting one or more `Endpoints` (similar
+to
+[Content Apps](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/media/VideoPlayerArchitecture.adoc#1-introduction)
+in the Matter Video Player architecture) that support one or more Matter Media
+`Clusters`.
+
+The steps to start a casting session are:
+
+1. [Initialize](#initialize-the-casting-client) the `CastingClient` using the
+   Matter SDK.
+1. [Discover](#discover-casting-players) `CastingPlayer` devices using Matter
+   Commissioner discovery.
+1. [Connect](#connect-to-a-casting-player) to the `CastingPlayer` to discover
+   available endpoints.
+1. [Select](#select-an-endpoint-on-the-casting-player) an available `Endpoint`
+   hosted by the `CastingPlayer`.
+
+Next, you're ready to:
+
+1. [Issue commands](#issuing-commands) to the `Endpoint`.
+1. [Read](#read-operations) endpoint attributes like playback state.
+1. [Subscribe](#subscriptions) to playback events.
+
+## Build and Setup
+
+The Casting Client is expected to consume the Matter TV Casting library built
+for its respective platform to consume any of the APIs described in this
+document. Refer to the tv-casting-app READMEs for [Android](android) and
+[iOS](darwin) to understand how to build and consume each platform's specicific
+libraries.
+
+### Initialize the Casting Client
+
+_{Complete Initialization examples:
+[Android](android/App/app/src/main/java/com/matter/casting/InitializationExample.java)
+| [iOS](darwin/TvCasting/TvCasting/TvCastingApp.swift)}_
+
+A Casting Client must first initialize the Matter SDK and define the following
+[`DataProvider`](android/App/app/src/main/jni/com/matter/casting/params/DataProvider.java)
+objects for the the Matter Casting library to use throughout the client's
+lifecycle:
+
+1.  **Rotating Device Identifier** - Refer to the Matter specification for
+    details on how to generate the
+    [Rotating Device Identifier](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/rendezvous/DeviceDiscovery.adoc#245-rotating-device-identifier)).
+    Then, instantiate a `DataProvider` object as described below.
+
+    On Android, define a `rotatingDeviceIdUniqueIdProvider` to provide the
+    Casting Client's RotatingDeviceIdUniqueId, by implementing a `DataSource`:
+
+    ```java
+    private final static DataProvider<byte[]> rotatingDeviceIdUniqueIdProvider = new DataProvider<byte[]>() {
+        private static final String APP_ID = "YOUR_GENERATED_ROTATING_DEVICE_ID";
```

Comment:
is APP_ID the best name for this? Also, shouldn't a real implementation pass the value in as an initializer argument given that it will likely be generated by the cloud or a random number generator?

Reply:
I've renamed it to ROTATING\_DEVICE\_ID\_UNIQUE\_ID to make what it is more obvious. 

About the initializer arg question - the DataProvider that provides this value is itself being passed as an initializer argument to the SDK.

Reply:
Should I add a comment saying that these are just sample values and you can pass any valid value to the SDK?

Reply:
Done